File: PokeApp/PokeApp/Models/Trainer.cs
﻿using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace PokeApp.Models
{
    public class Trainer
    {
        public int TrainerId { get; set; }
        [JsonProperty("userId")]
        public string UserId { get; set; }
        public string Name { get; set; }
        public int Caught { get; set; }

        public ICollection<Pokemon> Pokemons { get; set; }
        public ICollection<Team> Teams { get; set; }
    }
}

File: Frontend/src/app/models/trainer.ts
import { Team } from '../models/team';
import { PokemonInfo } from './pokemonInfo';

export class Trainer {
    constructor() {

    }

    trainerId: number;
    userId: string;
    name: string;
    caught: number;

    pokemons: PokemonInfo[];
    teams: Team[];
}
